Rename WLAN keyword args.

iabdalkader opened the pull request:
* Rename WLAN keyword args to connect() and config() to be more consistent across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword args, except for `essid` which is now deprecated in favor of `ssid`.
* Changed kwargs: `essid` changed to `ssid`, `auth` or `authmode` changed to `security` and `password` changed to `key`.

iabdalkader committed:
extmod/network_cyw43.c: Rename WLAN keyword args.

* Rename WLAN keyword args to connect() and config() to be more consistent
across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword
args, except for "essid" which is now deprecated in favor of "ssid".
* Changed kwargs: "essid" changed to "ssid", "auth" or "authmode" changed
to "security" and "password" changed to "key".

iabdalkader committed:
ports/esp32/network_wlan.c: Rename WLAN keyword args.

* Rename WLAN keyword args to connect() and config() to be more consistent
across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword
args, except for "essid" which is now deprecated in favor of "ssid".
* Changed kwargs: "essid" changed to "ssid", "auth" or "authmode" changed
to "security" and "password" changed to "key".

iabdalkader committed:
ports/esp8266: Rename WLAN keyword args.

* Rename WLAN keyword args to connect() and config() to be more consistent
across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword
args, except for "essid" which is now deprecated in favor of "ssid".
* Changed kwargs: "essid" changed to "ssid", "auth" or "authmode" changed
to "security" and "password" changed to "key".

iabdalkader committed:
docs: Rename WLAN keyword args.

* Rename WLAN keyword args to be more consistent across ports and wlan
drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword
args, except for "essid" which is now deprecated in favor of "ssid".
* Changed kwargs: "essid" changed to "ssid", "auth" or "authmode" changed
to "security" and "password" changed to "key".

dpgeorge commented on docs/library/network.WLAN.rst:
```diff
@@ -125,10 +125,10 @@ Methods
    =============  ===========
    mac            MAC address (bytes)
-   essid          WiFi access point name (string)
+   ssid          WiFi access point name (string)
```
please line up the second column

iabdalkader replied:
Fixed.

dpgeorge commented:
I think ninaw10 also needs some minor changes, for essid.

dpgeorge commented on docs/library/network.WLAN.rst:
```diff
@@ -125,10 +125,10 @@ Methods
-   authmode       Authentication mode supported (enumeration, see module constants)
+   hidden         Whether SSID is hidden (boolean)
+   security       Security protocol supported (enumeration, see module constants)
    password       Access password (string)
```
I think this needs changing to "key"

iabdalkader replied:
Replaced all the related `password` in the docs.

iabdalkader replied:
Note ESP ports use positional `ssid` and `key` not keyword args, but I fixed them anyway in their docs/quickref/tutorials.

iabdalkader committed:
extmod/network_ninaw10.c: Rename WLAN keyword args.

* Rename WLAN keyword args to connect() and config() to be more consistent
across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword
args, except for "essid" which is now deprecated in favor of "ssid".
* Changed kwargs: "essid" changed to "ssid", "auth" or "authmode" changed
to "security" and "password" changed to "key".

dpgeorge commented:
Rebased and merged in c502cf73e0e0df685723a828ecac1562e3be1cf0 through 6e868d47dc5dd7a90444f1173118b7d57e1e4df7
